src/testing/time_refinement_study_advection.cpp

```diff
@@ -0,0 +1,135 @@
+#include "time_refinement_study_advection.h"
+#include "flow_solver.h"
+#include "flow_solver_cases/periodic_1D_flow.h"
+#include "cmath"
+
+namespace PHiLiP {
+namespace Tests {
+
+template <int dim, int nstate>
+TimeRefinementStudyAdvection<dim, nstate>::TimeRefinementStudyAdvection(
+        const PHiLiP::Parameters::AllParameters *const parameters_input,
+        const dealii::ParameterHandler &parameter_handler_input)  
+        : TestsBase::TestsBase(parameters_input),
+         parameter_handler(parameter_handler_input)
+{}
+
+
+
+template <int dim, int nstate>
+ Parameters::AllParameters TimeRefinementStudyAdvection<dim, nstate>::reinit_params_and_refine_timestep(int refinement) const{
+     dealii::ParameterHandler parameter_handler;
+     PHiLiP::Parameters::AllParameters::declare_parameters (parameter_handler);
+     PHiLiP::Parameters::AllParameters parameters;
+     parameters.parse_parameters(parameter_handler);
+
+
+     //parameters = this->all_parameters; //This doesn't work, operator = not defined. Block below copies params one at a time.
```

Comment:
See how it's done in `src/testing/taylor_green_vortex_restart_check.cpp` function `reinit_params()`

Reply:
That is a much cleaner solution. I was referencing `fd_state_sensitivity_wrt_parameter.cpp` when writing this section. Will fix in next commit.

Reply:
Ok sounds good, just mark comments as "fixed" once the commit has actually been pushed, helps make the review easier